woodwork/tests/test_utils.py

```diff
@@ -566,3 +569,452 @@ def test_parse_logical_type_errors():
+
+
+def test_concat_cols_inner_join(sample_df):
+    df1 = sample_df.loc[[0, 1, 2], ['id', 'full_name', 'age']]
+    df1.ww.init(index='id')
+    df2 = sample_df.loc[[2, 3], ['email', 'signup_date']]
+    df2.ww.init(time_index='signup_date')
+
+    inner_combined = concat([df1, df2], join='inner', axis=1)
+    assert set(inner_combined.columns) == {'id', 'email', 'full_name', 'age', 'signup_date'}
+    assert len(inner_combined) == 1
+    assert inner_combined.ww.index == 'id'
+    assert inner_combined.ww.time_index == 'signup_date'
+
+
+def test_concat_rows_inner_join_lose_index(sample_df):
+    df1 = sample_df[['signup_date', 'email', 'age']]
+    df1.ww.init(time_index='signup_date')
+
+    df2 = sample_df[['id', 'email', 'signup_date', 'age']]
+    df2.ww.init(index='id', time_index='signup_date')
+
+    # If we lose a column in the inner join, we lose its typing information
+    inner_combined = concat([df1, df2], join='inner', axis=0)
+    assert set(inner_combined.columns) == {'email', 'age', 'signup_date'}
+    assert inner_combined.ww.index is None
+    assert inner_combined.ww.time_index == 'signup_date'
+
+
+def test_concat_rows_inner_join_lose_time_index(sample_df):
+    df1 = sample_df[['signup_date', 'email', 'age']]
+    df1.ww.init(time_index='signup_date')
+
+    df2 = sample_df[['id', 'email', 'age']]
+    df2.ww.init(index='id')
+
+    # If we lose a column in the inner join, we lose its typing information
+    inner_combined = concat([df1, df2], join='inner', axis=0)
+    assert set(inner_combined.columns) == {'email', 'age'}
+    assert inner_combined.ww.index is None
+    assert inner_combined.ww.time_index is None
+
+
+def test_concat_table_order(sample_df):
+    df1 = sample_df[['id', 'signup_date']]
+    df1.ww.init(index='id')
+    df2 = sample_df[['id', 'age']]
+    df2.ww.init(index='id')
+
+    combined_df1 = concat([df1.ww.copy(), df2.ww.copy()])
+    combined_df2 = concat([df2.ww.copy(), df1.ww.copy()])
+
+    assert set(combined_df1.columns) == set(combined_df2.columns)
+
+    assert list(combined_df1.columns) == ['id', 'signup_date', 'age']
+    assert list(combined_df2.columns) == ['id', 'age', 'signup_date']
+
+
+def test_concat_cols_all_series(sample_df):
+    age_series = ww.init_series(sample_df['age'], logical_type='Double')
+    combined_df = concat([sample_df['id'], age_series, sample_df['is_registered']], axis=1)
+
+    assert isinstance(combined_df.ww.schema, ww.table_schema.TableSchema)
+    assert set(combined_df.columns) == {'id', 'age', 'is_registered'}
+    assert combined_df.ww.logical_types['age'] == Double
+
+
+def test_concat_rows_all_series(sample_df):
+    warning = 'Woodwork not initialized on the output of ww.concat'
+    with pytest.warns(ww.exceptions.WoodworkNotInitWarning, match=warning):
+        combined_series = concat([sample_df['age'], sample_df['age']], axis=0)
+
+    assert combined_series.ww.schema is None
+    assert combined_series.name == 'age'
```

Comment:
Why can't we initialize Woodwork in this scenario?
```python
combined_series = concat([sample_df['age'], sample_df['age']], axis=0)
combined_series.ww.init()
combined_series.ww
```
```
<Series: age (Physical Type = Int64) (Logical Type = IntegerNullable) (Semantic Tags = {'numeric'})>
```

Reply:
We could, just didn't add support for it, so happy to clarify in the warning that it's not supported rather than making it seem like it's not possible.

Reply:
Situation no longer relevant